src/main/java/edu/depaul/cleanSweep/controlSystem/Cleaner.java

```diff
@@ -105,10 +111,18 @@ public FloorTile getCurrNode() {
 			this.changeHeading('N');
 		}
 		while(this.getCurrNode().get_y() != by) {
-			moveAhead();
+			if (moveAhead() == false)
+			{
+				if (blocked == true)
+				{
+					blocked = true;//both blocked in x and y direction
+					currStatus = "Blocked";
+				}
+				break;
+			}
 		}
 		this.changeHeading((char) dest[0]);
-		//System.out.println("^^^^^^^^^heading"+(char) dest[0]); //for test
+		System.out.println("^^^^^^^^^heading"+(char) dest[0]); //for test
```

Comment:
Something we should add to the main method is passing in a --debug option, which we treat as a boolean within the objects to enable/disable debug output.